darts/models/forecasting/query_selector.py

```diff
@@ -0,0 +1,650 @@
+
+        a = torch.stack(a, dim=-1)  # combine heads
+        a = a.flatten(start_dim=2)  # flatten all head outputs
+
+        x = self.fc(a)
+
+        return x
+
+    def record(self):
+        for h in self.heads:
+            h.record()
+
+
+class Value(nn.Module):
+    def __init__(self, dim_input, dim_val):
+        super().__init__()
+        self.dim_val = dim_val
+        self.fc1 = nn.Linear(dim_input, dim_val, bias=False)
+
+    def forward(self, x):
+        return self.fc1(x)
+
+
+class Key(nn.Module):
+    def __init__(self, dim_input, dim_attn):
+        super().__init__()
+        self.dim_attn = dim_attn
+        self.fc1 = nn.Linear(dim_input, dim_attn, bias=False)
+
+    def forward(self, x):
+        return self.fc1(x)
+
+
+class Query(nn.Module):
+    def __init__(self, dim_input, dim_attn):
+        super().__init__()
+        self.dim_attn = dim_attn
+        self.fc1 = nn.Linear(dim_input, dim_attn, bias=False)
+
+    def forward(self, x):
+        return self.fc1(x)
+
+
+class PositionalEncoding(nn.Module):
+    def __init__(self, d_model, dropout=0.1, max_len=5000):
+        super().__init__()
+        self.dropout = MonteCarloDropout(p=dropout)
+
+        pe = torch.zeros(max_len, d_model)
+        position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
+
+        div_term = torch.exp(
+            torch.arange(0, d_model, 2).float() * (-math.log(10000.0) / d_model)
+        )
+
+        pe[:, 0::2] = torch.sin(position * div_term)
+        pe[:, 1::2] = torch.cos(position * div_term)
+
+        pe = pe.unsqueeze(0).transpose(0, 1)
+
+        self.register_buffer("pe", pe)
+
+    def forward(self, x):
+        x = x + self.pe[: x.size(0), :]
+        return self.dropout(x)
+
+
+def FeedForward(d_model, d_ff, dropout=0.1):
+    return nn.Sequential(
+        nn.Linear(d_model, d_ff),
+        nn.Sigmoid(),
+        MonteCarloDropout(p=dropout),
+        nn.Linear(d_ff, d_model),
+        MonteCarloDropout(p=dropout),
+    )
+
+
+class FeedForwardBlock(nn.Module):
+    def __init__(self, d_model: int, d_ff: int, **kwargs):
+        super().__init__()
+        self.norm = nn.LayerNorm(d_model)
+        self.ff = FeedForward(d_model, d_ff, **kwargs)
+        self.post_norm = nn.LayerNorm(d_model)
+
+        init_function = partial(init_weights)
+        self.ff.apply(init_function)
+
+    def forward(self, x):
+        x = self.norm(x)
+        return self.post_norm(x + self.ff(x))
+
+
+def init_weights(module):
+    if type(module) == torch.nn.Linear:
+        torch.nn.init.xavier_uniform_(module.weight)
+
+
+class EncoderLayer(nn.Module):
+    def __init__(
+        self,
+        dim_val,
+        dim_attn,
+        feed_forward,
+        n_heads=1,
+        attn_type="full",
+        norm=nn.LayerNorm,
+    ):
+        super().__init__()
+        self.attn = MultiHeadAttentionBlock(
+            dim_val, dim_attn, n_heads, attn_type=attn_type
+        )
+
+        self.norm1 = norm(dim_val)
```

Comment:
These normalization don't have parameters, so I don't think you need to define two of them, one should be enough.